README.md

```diff
@@ -1,23 +1,108 @@
-To start your Phoenix server:
+Proxy server to facilitate communication with Edgehog devices.
+It handles HTTP messages and WebSocket connections between the device and the user client.
 
-  * Run `mix setup` to install and setup dependencies
-  * Start Phoenix endpoint with `mix phx.server` or inside IEx with `iex -S mix phx.server`
+It's only been tested with simple WebSocket connections such as
+[ttyd](https://github.com/tsl0922/ttyd), however it should handle arbitrary requests
+as long as you keep in mind the [limitations](#known-issues-and-limitations).
 
-Now you can visit [`localhost:4000`](http://localhost:4000) from your browser.
+## Running the Forwarder locally
 
-Ready to run in production? Please [check our deployment guides](https://hexdocs.pm/phoenix/deployment.html).
+### Start the server
 
-## Learn more
+The first component to be available should be this server,
+as both the device and the client will connect to it.
 
-  * Official website: https://www.phoenixframework.org/
-  * Guides: https://hexdocs.pm/phoenix/overview.html
-  * Docs: https://hexdocs.pm/phoenix
-  * Forum: https://elixirforum.com/c/phoenix-forum
-  * Source: https://github.com/phoenixframework/phoenix
+#### Manual installation
+
+Run the following command to clone the current repository and navigate to it
+
+```bash
+git clone https://github.com/edgehog-device-manager/edgehog_device_forwarder && \
+cd edgehog_device_forwarder
+```
+
+> At the time of writing, not all the relevant code has been merged into the main branch.
+> Use the following command command instead
+>
+> ```bash
+> git clone https://github.com/noaccOS/edgehog_device_forwarder \
+> -b feat/fallback-controller && cd edgehog_device_forwarder
+> ```
+
+You will need to install Elixir 1.15 with OTP 26.
+
+> If you use [`asdf`](https://asdf-vm.com), you can use it to install our officially supported
+> version, by using the `.tool-versions` file from
+> [edgehog's repository](https://github.com/edgehog-device-manager/edgehog/blob/main/.tool-versions)
+> and copy it here.
+
+Now use the following command to start the server:
+
+```bash
+mix phx.server
+```
+
+### Connect a device
+
+You will now need to start the device process.
+You can do so from the same machine or another device or virtual machine, the process should look
+exactly the same (granted port 4000 is open on the server's firewall).
+
+To connect to the server you will need to use the e2e-test-forwarder program, available in the
+[`edgehog-device-runtime`](https://github.com/edgehog-device-manager/edgehog-device-runtime)
+repository.
+
+#### Manual install
+
+> At the time of writing, you need to use a fork of said repository, because not all the code
+> has been merged.
+> Use the command
+>
+> ```bash
+> git clone https://github.com/rgallor/edgehog-device-runtime -b ws-over-ws && \
+> cd edgehog-device-runtime/e2e-test-forwarder
+> ```
+
+This is a Rust program, so to compile it you will need to install `cargo`. Then run it using
```

Comment:
Is it useful to link [this](https://doc.rust-lang.org/cargo/getting-started/installation.html)?

Reply:
I opted for [this one](https://www.rust-lang.org/tools/install) instead (was mentioned in your link)